### PR TITLE
Added simple_ga.py algo file

### PR DESCRIPTION
**Pull request by MaximilienLC**

### CODE REVIEW

File: evojax/algo/simple_ga.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self.params = self.tell_fn(fitness, self.params)
+        self._best_params = self.params[-1]
+
+    @property
+    def best_params(self):
+        return self.jnp_array(self._best_params)
+
+    @best_params.setter
+    def best_params(self, params):
+        self._best_params = params
```
Comment (lerrytang) [inline evojax/algo/simple_ga.py:62]: By setting `best_params`, we expect the algorithm can continue training from that point, it seems `self._best_params` is not used in this way.

**Comment by lerrytang** [comment]
Hi, thanks for the PR, I'm testing its performance on the tasks. 
At the same time, can you take a look at my review comment and make changes accordingly?
Please see our [implementation](https://github.com/google/evojax/blob/main/evojax/algo/pgpe.py) as an example.

File: evojax/algo/simple_ga.py (hunk 1 above)
Comment (MaximilienLC) [reply evojax/algo/simple_ga.py]: Should work as intended now.

Comment (MaximilienLC) [reply evojax/algo/simple_ga.py]: Actually I'm a little confused about the format of `best_params`. Is it the parameters of the elite agent or the whole batch of parameters from the entire population?

Comment (lerrytang) [reply evojax/algo/simple_ga.py]: It should be the top elite agent's parameter.
Users call `NEAlgorithm.best_params` to save/test the model.

Comment (MaximilienLC) [reply evojax/algo/simple_ga.py]: Got it and fixed the setter!

**Comment by lerrytang** [comment]
**Test results**

|                 | Benchmarks           | Parameters            | Results |
|-----------------|----------------------|-----------------------|---------|
| MNIST           | 90.0 (max_iter=5000) | sigma=0.001           | 90.8    |
| CartPole (easy) | 900 (max_iter=2000)  | default               | 925     |
| CartPole (hard) | 600 (max_iter=2000)  | default               | 616     |
| Waterworld      | 6 (max_iter=2000)    | default               | 6.39    |
| Waterworld (MA) | 2 (max_iter=5000)    | default               | 1.19    |
| Brax Ant        | 3000 (max_iter=1000) | truncation_divisor= 4 | 2336    |

**Notes**

1. This table shows the implementation's performance so that users can choose the algorithms for their experiments.
3. We are aware that some algorithms have limitations (e.g., unable to train large policy network), the benchmarks are therefore not hard requirements. However, we refuse to merge if some scores are significantly lower.
4. After we release the test scripts, the PR submitter will be responsible for producing this table. We can help fill in some entries if the submitter cannot run the experiments due to hardware limitations.

**Pull request merged**